test/domain.test.ts

```diff
@@ -2,18 +2,60 @@ import 'mocha';
 import { expect } from 'chai';
 import * as domain from 'domain';
 import { EventEmitter } from 'events';
+import { createRequest, createResponse } from 'node-mocks-http';
 
 import * as globalStore from '../src';
+import Middleware from '../src/Middleware';
 import { STORE_KEY } from '../src/StoreDomain';
 import AsyncStoreAdapter from '../src/AsyncStoreAdapter';
 
 describe('store: [adapter=DOMAIN]', () => {
   const adapter = AsyncStoreAdapter.DOMAIN;
+  const initMiddleware: Middleware = globalStore.initializeMiddleware(adapter);
+  const initDefaultMiddleware: Middleware = globalStore.initializeMiddleware();
 
   beforeEach(() => {
     Object.assign(process, { domain: undefined });
   });
 
+  describe('initializeMiddleware()', () => {
+    it('should return a function which references the middleware', () => {
+      expect(initMiddleware).to.be.a('function');
+      expect(initDefaultMiddleware).to.be.a('function');
+    });
+
+    it('should return 3 arguments', () => {
+      expect(initMiddleware.length).to.equal(3);
+      expect(initDefaultMiddleware.length).to.equal(3);
+    });
+  });
+
+  describe('isInitialized() through middleware', () => {
+    const [req, res] = [createRequest(), createResponse()];
+
+    function checkStore(done: Mocha.Done) {
```

Comment:
Define arrow function to define functions inside a function. Only for the root level functions should we use `function` keyword.

```suggestion
    const checkStore = (done: Mocha.Done) => {
```

Reply:
Got it